File: 3_huli/exams/partner_tests/ejogseged_test/tasks_by_lang/JAVA/Traversal.java
package hu.ejogseged.challenge;

import java.util.*;
import org.apache.commons.lang3.ArrayUtils;

//
// Készítsd el a megadott fa adatszerkezetet bejáró metódusokat:
//
// Példa kód:
//
// HashMap<String, Object> tree1 = new HashMap<String, Object>();
// tree1.put("value", 1);
// HashMap<String, Object> tree2 = new HashMap<String, Object>();
// tree2.put("value", 2);
// HashMap<String, Object> tree3 = new HashMap<String, Object>();
// tree3.put("value", 3);
// HashMap<String, Object> tree4 = new HashMap<String, Object>();
// tree4.put("value", 4);
// HashMap<String, Object> tree5 = new HashMap<String, Object>();
// tree5.put("value", 5);
// HashMap<String, Object> tree6 = new HashMap<String, Object>();
// tree6.put("value", 6);
// HashMap<String, Object> tree7 = new HashMap<String, Object>();
// tree7.put("value", 7);
// ArrayList<HashMap<String, Object>> childrenOfTree1 = new ArrayList<HashMap<String, Object>>();
// childrenOfTree1.add(tree2);
// childrenOfTree1.add(tree3);
// tree1.put("children", childrenOfTree1);
// ArrayList<HashMap<String, Object>> childrenOfTree2 = new ArrayList<HashMap<String, Object>>();
// childrenOfTree2.add(tree4);
// tree2.put("children", childrenOfTree2);
// ArrayList<HashMap<String, Object>> childrenOfTree3 = new ArrayList<HashMap<String, Object>>();
// childrenOfTree3.add(tree5);
// childrenOfTree3.add(tree6);
// tree3.put("children", childrenOfTree3);
// ArrayList<HashMap<String, Object>> childrenOfTree4 = new ArrayList<HashMap<String, Object>>();
// tree4.put("children", childrenOfTree4);
// ArrayList<HashMap<String, Object>> childrenOfTree5 = new ArrayList<HashMap<String, Object>>();
// tree5.put("children", childrenOfTree5);
// ArrayList<HashMap<String, Object>> childrenOfTree6 = new ArrayList<HashMap<String, Object>>();
// childrenOfTree6.add(tree7);
// tree6.put("children", childrenOfTree6);
// ArrayList<HashMap<String, Object>> childrenOfTree7 = new ArrayList<HashMap<String, Object>>();
// tree7.put("children", childrenOfTree7);
//
// Példa ábra:
//
//                             +----------------+
//                             |     value:   1 |
//                             +----------------+
//                               /            \
//                              /              \
//               +----------------+        +----------------+
//               |     value:   2 |        |     value:   3 |
//               +----------------+        +----------------+
//                  /                         /           \
//                 /                         /             \
//    +----------------+       +----------------+       +----------------+
//    |     value:   4 |       |     value:   5 |       |     value:   6 |
//    +----------------+       +----------------+       +----------------+
//                                                           /
//                                                          /
//                                            +----------------+
//                                            |     value:   7 |
//                                            +----------------+
//

public class Traversal
{
    //
    // Pre-order bejárás eredménye a példa adatokra:  [ 1, 2, 4, 3, 5, 6, 7 ]
    //
    public int[] preOrderTraversal(HashMap<String, Object> tree)
    {
        return null; // TODO
    }

    //
    // Post-order bejárás eredménye a példa adatokra: [ 4, 2, 5, 7, 6, 3, 1 ]
    //
    public int[] postOrderTraversal(HashMap<String, Object> tree)
    {
        return null; // TODO
    }
}

File: 3_huli/exams/partner_tests/ejogseged_test/tasks_by_lang/C#/Traversal.cs
using System;
using System.Collections.Generic;

//
// Készítsd el a megadott fa adatszerkezetet bejáró metódusokat:
//
// Példa kód:
//
//Dictionary<string, object> tree1 = new Dictionary<string, object>();
//tree1.Add("value", 1);
//Dictionary<string, object> tree2 = new Dictionary<string, object>();
//tree2.Add("value", 2);
//Dictionary<string, object> tree3 = new Dictionary<string, object>();
//tree3.Add("value", 3);
//Dictionary<string, object> tree4 = new Dictionary<string, object>();
//tree4.Add("value", 4);
//Dictionary<string, object> tree5 = new Dictionary<string, object>();
//tree5.Add("value", 5);
//Dictionary<string, object> tree6 = new Dictionary<string, object>();
//tree6.Add("value", 6);
//Dictionary<string, object> tree7 = new Dictionary<string, object>();
//tree7.Add("value", 7);
//List<Dictionary<string, object>> childrenOfTree1 = new List<Dictionary<string, object>>();
//childrenOfTree1.Add(tree2);
//childrenOfTree1.Add(tree3);
//tree1.Add("children", childrenOfTree1);
//List<Dictionary<string, object>> childrenOfTree2 = new List<Dictionary<string, object>>();
//tree2.Add("children", childrenOfTree2);
//List<Dictionary<string, object>> childrenOfTree3 = new List<Dictionary<string, object>>();
//childrenOfTree3.Add(tree5);
//childrenOfTree3.Add(tree6);
//tree3.Add("children", childrenOfTree3);
//List<Dictionary<string, object>> childrenOfTree4 = new List<Dictionary<string, object>>();
//tree4.Add("children", childrenOfTree4);
//List<Dictionary<string, object>> childrenOfTree5 = new List<Dictionary<string, object>>();
//tree5.Add("children", childrenOfTree5);
//List<Dictionary<string, object>> childrenOfTree6 = new List<Dictionary<string, object>>();
//childrenOfTree6.Add(tree7);
//tree6.Add("children", childrenOfTree6);
//List<Dictionary<string, object>> childrenOfTree7 = new List<Dictionary<string, object>>();
//tree7.Add("children", childrenOfTree7);
//
// Példa ábra:
//
//                             +----------------+
//                             |     value:   1 |
//                             +----------------+
//                               /            \
//                              /              \
//               +----------------+        +----------------+
//               |     value:   2 |        |     value:   3 |
//               +----------------+        +----------------+
//                  /                         /           \
//                 /                         /             \
//    +----------------+       +----------------+       +----------------+
//    |     value:   4 |       |     value:   5 |       |     value:   6 |
//    +----------------+       +----------------+       +----------------+
//                                                           /
//                                                          /
//                                            +----------------+
//                                            |     value:   7 |
//                                            +----------------+
//

public class Travelsal
{
    //
    // Pre-order bejárás eredménye a példa adatokra:  [ 1, 2, 4, 3, 5, 6, 7 ]
    //
    public int[] preOrderTraversal(Dictionary<string, object> tree)
    {        
        return null; // TODO
    }

    //
    // Post-order bejárás eredménye a példa adatokra: [ 4, 2, 5, 7, 6, 3, 1 ]
    //
    public int[] postOrderTraversal(Dictionary<string, object> tree)
    {
        return null; // TODO
    }
}